pkg/p2p/libp2p/internal/handshake/handshake.go

```diff
@@ -271,6 +278,18 @@ func (s *Service) Handle(ctx context.Context, stream p2p.Stream, remoteMultiaddr
 		s.logger.Infof("greeting \"%s\" from peer: %s", ack.WelcomeMessage, remoteBzzAddress.Overlay.String())
 	}
 
+	incomingTx := common.HexToHash(ack.Transaction)
+
+	rcpt, err := s.swapBackend.TransactionReceipt(context.Background(), incomingTx)
+	if err != nil {
+		return nil, fmt.Errorf("get transaction receipt: %w", err)
+	}
+
+	expectedRemoteBzzAddress := crypto.NewOverlayFromEthereumAddress(rcpt.ContractAddress.Bytes(), s.networkID)
+	if !expectedRemoteBzzAddress.Equal(remoteBzzAddress.Overlay) {
+		return nil, fmt.Errorf("given address is spoofed: %v", remoteBzzAddress.Overlay)
```

Comment:
`spoofed` is jargon and does not necessarily say something to the user. Consider: `given address is not registered on Ethereum`

Reply:
good point, I'll update it